ui/app/components/not-found.js

```diff
@@ -20,7 +20,16 @@ import Component from '@glimmer/component';
 export default class NotFound extends Component {
   @service router;
 
-  get path() {
-    return this.router.currentURL;
+  constructor() {
+    super(...arguments);
+    const { currentURL } = this.router;
+    if (currentURL.startsWith('/vault/settings/secrets/configure/')) {
```

Comment:
My only hesitation with this is if a user is trying to get to the configure page of a backend that doesn't exist, we might get into an infinite loop of redirects. I might prefer that we keep the old route for now, show a flash message on redirect alerting the user that the route will be removed in future versions, and then have a deprecation schedule for the route. 
EDIT: I checked this locally, and it looks like attempting to access a backend that doesn't exist errors with a 403 instead of a 404 so the not-found template doesn't render, which is good. May still be worth evaluating our route deprecation approach generally because this has come up before and we didn't really have a cohesive plan